### DISCUSSION TITLE
How can I specify in the configuration file or command line that my simulation can wait for remote GDB connections before starting simulation

### DISCUSSION THREAD

**Discussion by flyeagle0** [gem5/gem5#823] [Q&A] [answered]
**when I run this command in my command line :**
$ ./build/RISCV/gem5.opt myconfigfile.py
and the following output was obtained
build/RISCV/arch/riscv/linux/se_workload.cc:60: warn: Unknown operating system; assuming Linux.
0: system.remote_gdb: listening for remote gdb on port 7000
#########################......Omit partial output.......##############################
**I guess Gem5 has a built-in gdb server that allows me to connect to Gem5 using gdb to debug the program I simulate on Gem5, rather than using gdb to connect to Gem5 to debug it itself。 Because I know that there is a mechanism in Spike emulators, such as**
$ spike --rbb-port=9824 -m0x10100000:0x20000 rot13-64
Listening for remote bitbang connection on port 9824.
$ openocd -f spike.cfg
$ riscv64-unknown-elf-gdb rot13-64
GNU gdb (GDB) 8.0.50.20170724-git
#########################......Omit partial output.......##############################
Reading symbols from rot13-64...done.
(gdb) target remote localhost:3333
and then I can debug my program running on Spike now

**I saw that the output of GEM5 has a GDB listening port, so I guess GEM5 integrates a GDB server, but I haven't found a way to stop my program so that I can connect to GEM5 using GDB and debug my program**

Does anyone know what to do? Thanks!

**Comment by rogerchang23424** [comment, marked as answer]
It depends on the type of the myconfig.py

If the config file is like `example/se.py` and `example/fs.py`. You can add `--wait-gdb` to let gem5 pause simulation until the GDB is connected.

If the user run simulation with `RiscvBoard`. You may need to modify the `RiscvBoard._setup_board` and rebuild again to let gem5 enable wait for gdb connection

```python
    @overrides(AbstractSystemBoard)
    def _setup_board(self) -> None:
        self.workload = RiscvBootloaderKernelWorkload()
        
        self.wait_for_remote_gdb = True  # Add the code to let gem5 wait for remote GDB connection
```

**Reply by flyeagle0** [reply]
Thank you a lot! My configuration file references Learning Gem5, and I have added the following statement to it
`system.workload.wait_for_remote_gdb = True`
and completed my testing by following these steps
```
$ gem5.opt configfile.py
..................................
Beginning simulation!
src/sim/workload.cc:101: info: system.workload: Waiting for a remote GDB connection on port 7000.
```
open another terminal
```
$ riscv32-unknown-elf-gdb yourprogram.elf
$(gdb) target remote localhost:7000
Remote debugging using localhost:7000
main () at ./script/prog/example_prog.c:19
```
